submission/proxy.go

```diff
@@ -168,8 +168,8 @@ func (p *Proxy) restartDistributor(ctx context.Context, ll *loglist2.LogList) er
 	refreshCtx, refreshCancel := context.WithCancel(ctx)
 	go schedule.Every(refreshCtx, p.rootsRefreshInterval, func(ectx context.Context) {
 		if errs := d.RefreshRoots(ectx); len(errs) > 0 {
```

Comment:
Consider doing a similar clean-up with the channel returned from `d.RefreshRoots`.